docs/snapshotting/versioning.md

```diff
@@ -40,41 +60,51 @@ A microVM state file is further split into four different fields:
 "Versionize layers")
 
-## Bincode encoding
-During research and prototyping we have considered multiple storage formats. The criteria used for comparing these were performance, size, Rust support, specification, versioning support, community popularity and tooling. Performance, size and Rust support are hard requirements while all others can be the subject of trade-offs. 
+## VM state encoding
+
+During research and prototyping we considered multiple storage formats. The criteria used for comparing these are: performance, size, rust support, specification, versioning support, community and tooling. Performance, size and Rust support are hard requirements while all others can be the subject of trade offs.
+More info about this comparison can be found here (https://github.com/firecracker-microvm/firecracker/blob/9d427b33d989c3225d874210f6c2849465941dc0/docs/snapshotting/design.md).
+
+Key benefits of using *bincode* versus everything else:
 
-More info about this comparison can be found [here](https://github.com/firecracker-microvm/firecracker/blob/9d427b33d989c3225d874210f6c2849465941dc0/docs/snapshotting/design.md#we-also-looked-at-these-other-options).
+* Minimal snapshot size overhead
+* Minimal CPU overhead
+* Simple implementation
 
-Key benefits of using `bincode` versus everything else are:
-- Minimal snapshot size overhead,
-- Minimal CPU overhead,
-- Simple implementation.
+The current implementation relies on the  Serde bincode (https://github.com/servo/bincode) for the signed and unsigned integer primitive types and provides a custom implementation for higher level primitives like vectors and strings. We have also added extra checks for variable field lengths. 
+
+Versionize is compatible to Serde with bincode backend: structures serialized with versionize at a specific version can be deserialized with Serde. Also structures serialized with serde can be deserialized with versionize.
 
 ## Snapshot compatibility
 ### Host kernel
 
-The minimum kernel version required by Firecracker snapshots is 4.14. Snapshots can be saved and restored on different kernel versions without issues.
+The minimum kernel version required by Firecracker snapshots is 4.14. Snapshots can be saved and restored on the same kernel version without any issues. There might be issues when restoring snapshots created on different host kernel version even when using the same Firecracker version. 
+
+SnapshotCreate and SnapshotLoad operations across different host kernels is considerted unstable in Firecracker as the saved KVM state might have different semantics on a different kernels.
+
 
 ### Device model
+The current Firecracker devices are backwards compatible up to the version that introduces them. Ideally this property would be kept over time, but there are situations when a new version of a device exposes new features to the guest that do not exist in an older version. In such cases rollback becomes impossible without breaking the guest workload.
```

Comment:
I'm guessing this depends on whether the guest workload is actually using the feature, right?

Reply:
We cannot trust the guest, we trust the microvm state. We simply disallow the creation of such snapshots or prevent them from being loaded.